cloud-storage-reference-app.md

```diff
@@ -0,0 +1,366 @@
+    Patient Peter CH App->>Patient Peter CH App: Periodic Vendor Data Download
+    Patient Peter CH App->>CHCS-IAM Web App: Uploads Data with<br/>OAuth2 Access Token
+    Researcher Rachel->>CHCS-IAM Web App: Views/sorts/downloads<br/>Patient Peter data
+```
+
+## SMART Client Flow
+
+```mermaid
+sequenceDiagram
+    actor Dr Debora
+    participant EHR
+    participant SMART CHCS
+    Dr Debora->>EHR: Seaches Patient Peter<br/>Clicks to launch CH
+    EHR->>SMART CHCS: GET chcs.org/launch<br/>?iss=https://fhir.example.com&launch=xyz123
+    SMART CHCS->>EHR: GET fhir.ehr.com/.well-known/smart-configuration
+    EHR->>SMART CHCS: Conformance JSON<br/>grant_types_supported, authorization_endpoint,<br/>jwks_uri, token_endpoint, capabilities,<br/>code_challenge_methods_supported
+    SMART CHCS->>EHR: GET auth.ehr.com<br/>?client_id=abc123&launch=xyz123<br/>&state=999&scope=launch+patient<br/>&redirect_uri=https://chcs.org/redirect&...
+    EHR->>SMART CHCS: GET chcs.org/redirect<br/>?code=1111&state=999
+    SMART CHCS->>EHR: POST<br/>code=1111, grant_type=access_token,scope=launch+patient,<br/>code_verifier=0f01, redirect_uri=https://chcs.org/redirect
+    EHR->>SMART CHCS: Auth JSON<br/>access_token, token_type, scope, patient=def123.456
+    SMART CHCS->>EHR: GET /fhir/patients/def.456<br/>Bearer: access_token
+    SMART CHCS->>SMART CHCS: Matches EHR user.id<br/>to CHCS user
+    EHR->>SMART CHCS: First Name, Last Name, Cell, E-mail
+    SMART CHCS->>SMART CHCS: Matches EHR patient.id<br/>to CHCS patient
+    SMART CHCS->>SMART CHCS: (Continues flow with Patient Peter<br/>SMART CHCS as above)
+    SMART CHCS->>Dr Debora: Dr Debora signed in<br/>Patient Peter shared personal device data displayed
+```
+
+______________________________________________________________________
+
+# Reference Material
+
+## Considerations
+
+### Cloud Storage Service(CSS) as a SMART Server
+
+- SMART Cloud Storage Service is source of truth for auth (with exception of SAML/SSO)
+- For access control against personal device data Cloud Storage Service should also be store of records, otherwise will require double-auth or some kind of pre-signed URIs and additional complexity unnecessary for a reference implementation
+- Without Secondary App access, auth is really just OAuth2 and not SMART
+
+```mermaid
+sequenceDiagram
+    actor Dr Debora
+    participant SMART CSS
+    participant Provider Directory Service
+    participant Secondary App (SMART Client)
+    Dr Debora->>SMART CSS: css.org<br/>Sign up / Sign in
+    opt
+        SMART CSS->>Provider Directory Service: SAML/SSO Auth
+        Provider Directory Service->>SMART CSS:
+    end
+    Dr Debora->>SMART CSS: Set up study, users and groups
+    Dr Debora->>SMART CSS: Register Patient, create and send deep link
+    Dr Debora->>Secondary App (SMART Client): Click to launch
+    Secondary App (SMART Client)->>SMART CSS: SMART Auth flow
+    SMART CSS->>Secondary App (SMART Client):
```

Comment:
Same as above -- MyST wants something following `(SMART Client): `.

Reply:
@s1monj I am not sure what to do here since if I change it then I get an error. I guess we could just screen capture as an image?